runtime/lua/vim/shared.lua

```diff
@@ -1162,6 +1162,32 @@ function vim._defer_require(root, mod)
   })
 end
 
+--- @private
+--- Creates a module alias/shim that lazy-loads a target module.
+---
+--- Unlike `vim.defaulttable()` this also:
+--- - implements __call
+--- - calls vim.deprecate()
+---
+--- @param old_name string Name of the deprecated module, which will be shimmed.
+--- @param new_name string Name of the new module, which will be loaded by require().
+function vim._defer_deprecated_module(old_name, new_name)
```

Comment:
I rushed this, will iterate on it later. Though if anyone sees an obvious problem, LMK.